### PR TITLE
Code refactor completed.

### PR DESCRIPTION
**Pull request by laskarj**

### CODE REVIEW

**Review by Grayder0152 — CHANGES_REQUESTED** [review]
Pls resolve issue
image

File: app/models/knight.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+class Knight:
+
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: List[Armour],
+            weapon: Weapon,
+            potion: Potion
```
Comment (vsmutok) [inline app/models/knight.py:15]: it is better to take the dictionary right away

File: app/models/knight.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = 0
+
+    def preparation(self) -> None:
+        if self.armour:
+            for arm in self.armour:
+                self.protection += arm.protection
```
Comment (vsmutok) [inline app/models/knight.py:28]: try to use `sum`

File: app/models/knight.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        if self.weapon:
+            self.power += self.weapon.power
+
+        if self.potion:
+            drank = self.potion.drink()
+            self.hp += drank[0]
+            self.power += drank[1]
+            self.protection += drank[2]
+
+    def __isub__(self, other: Callable) -> Callable:
```
Comment (vsmutok) [inline app/models/knight.py:39]: Maybe `duel`  better name

File: app/models/knight.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                self.protection += arm.protection
+
+        if self.weapon:
+            self.power += self.weapon.power
+
+        if self.potion:
+            drank = self.potion.drink()
+            self.hp += drank[0]
+            self.power += drank[1]
+            self.protection += drank[2]
```
Comment (vsmutok) [inline app/models/knight.py:37]: use unpacking here

File: app/models/tool_model.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+class Potion:
+
+    def __init__(self, name: str, effect: dict) -> None:
+        self.name = name
+        self.effect = effect
+
+    def drink(self) -> tuple:
+        hp = self.effect["hp"] if "hp" in self.effect.keys() else 0
+        power = self.effect["power"] if "power" in self.effect.keys() else 0
+        protection = (
```
Comment (vsmutok) [inline app/models/tool_model.py:24]: use `get`

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Pull request closed without merging**